### ISSUE TITLE
[Feature Request] Geolocation/GPS "spoofing" on android

### ISSUE THREAD

**Issue by NewRedsquare** [mullvad/mullvadvpn-app#2119]
## Issue report

Operating system: Android

### Feature description

I see many VPN provider apps that have a feature to spoof the GPS so it matches the country where the client is connected to. For instance, if i'm connected to Switzerland, the GPS will show that i'm in Zurich.
This feature would complete the overall privacy of the device.

**Comment by faern** [comment]
Thanks for the suggestion. But the only way this can be done, to my knowledge, is with either a rooted phone or if the phone is running in developer mode. Neither of which we want to recommend our users to do.

This feature sounds somewhat out of scope for a VPN app. This functionality could very well be implemented in a standalone app that spoofs the GPS position to the same place as the device's public IP (which would be the VPN server if the device is connected via VPN)

Do you have a practical use case for when this would be a useful feature? Personally I block apps that I don't trust with my position from getting my position. And for apps I want to have my position, I usually want them to have the correct one in order for them to function well. Take for example a public transport app, I don't want it to suggest rides from Zurich if I'm in Berlin :)

**Comment by 8227846265** [comment]
I agree with faern said, this is impossible for the Android system unless it's a standalone app and the device is rooted/developer mode.

GPS on smartphones is powered by a hardware inside your smartphone. It's not completely software-based. The software in addition is already configured to do certain things unless tampered to something else (which is why you need to be rooted to tamper to these configurations)

There are however, some apps that can do this without root however you must disable high accuracy location positioning settings or more accurately. Enable Developer Mode and enable "Allow Mock Locations". 

In the end though, setting it up on Android is too much of a hassle. The mere usage of these drains the battery very fast.

**As for the statement:**
"I see many VPN provider apps that have a feature to spoof the GPS"

I have never of many VPN provider that does this. From my knowledge only Windscribe with their Browser Extension is able to do this.

**Comment by NewRedsquare** [comment]
Windscribe does natively with its android app

**Comment by firepacket** [comment]
Just enable developer mode and you can set your GPS location. Mullvad does not need to waste time on this, and there are other more robust solutions.

**Comment by Rawa** [comment]
I'm going to close this as out of scope for the android app, and that it would be very niche since most people do not run developer options. This is also detectable by other apps, so allowing it would not be foolproof in any way.
